src/sliceDataStorage.cpp

```diff
@@ -469,4 +477,31 @@ bool SliceDataStorage::getExtruderPrimeBlobEnabled(int extruder_nr) const
     return train->getSettingBoolean("prime_blob_enable");
 }
 
+
+void SupportLayer::excludeAreasFromSupportInfillAreas(const Polygons& exclude_polygons, const AABB& exclude_polygons_boundary_box)
+{
+    // take the differences of the support infill parts and the exclude polygons
+    for (auto part_itr = this->support_infill_parts.begin(); part_itr != this->support_infill_parts.end(); ++part_itr)
```

Comment:
This function has a worst case running time of O(n^2).

I think this can be handled better.
Let's discuss this some more tomorrow.